feature/data-generator/src/main/java/com/simprints/feature/datagenerator/DataGeneratorViewModel.kt

```diff
@@ -0,0 +1,147 @@
+package com.simprints.feature.datagenerator
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.simprints.feature.datagenerator.enrollmentrecords.InsertEnrollmentRecordsUseCase
+import com.simprints.infra.authstore.AuthStore
+import com.simprints.infra.logging.Simber
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+internal class DataGeneratorViewModel @Inject constructor(
+    private val insertEnrollmentRecords: InsertEnrollmentRecordsUseCase,
+    private val authStore: AuthStore,
+) : ViewModel() {
+    companion object {
+        private const val TAG = "DataGeneratorViewModel"
+
+        // Intent Actions
+        private const val ACTION_GENERATE_ENROLLMENT_RECORDS = "com.simprints.test.GENERATE_ENROLLMENT_RECORDS"
+        private const val ACTION_GENERATE_SESSION_EVENTS = "com.simprints.test.GENERATE_SESSION_EVENTS"
+
+        // Common Extras
+        private const val EXTRA_PROJECT_ID = "EXTRA_PROJECT_ID"
+        private const val EXTRA_MODULE_ID = "EXTRA_MODULE_ID"
+        private const val EXTRA_ATTENDANT_ID = "EXTRA_ATTENDANT_ID"
+
+        // Enrollment Record Extras
+        private const val EXTRA_NUM_RECORDS = "EXTRA_NUM_RECORDS"
+        private const val EXTRA_TEMPLATES_PER_FORMAT = "EXTRA_TEMPLATES_PER_FORMAT"
+        private const val EXTRA_FIRST_SUBJECT_ID = "EXTRA_FIRST_SUBJECT_ID"
+        private const val EXTRA_FINGER_ORDER = "EXTRA_FINGER_ORDER"
+
+        // Session Event Extras
+        private const val EXTRA_ENROL_COUNT = "EXTRA_ENROL_COUNT"
+        private const val EXTRA_IDENTIFY_COUNT = "EXTRA_IDENTIFY_COUNT"
+        private const val EXTRA_CONFIRM_IDENTIFY_COUNT = "EXTRA_CONFIRM_IDENTIFY_COUNT"
+        private const val EXTRA_ENROL_LAST_COUNT = "EXTRA_ENROL_LAST_COUNT"
+        private const val EXTRA_VERIFY_COUNT = "EXTRA_VERIFY_COUNT"
+    }
+
+    private val _statusMessage = MutableLiveData<String>()
+    val statusMessage: LiveData<String> = _statusMessage
+
+    suspend fun handleIntent(intent: Intent?) {
+        Simber.i("Handling intent: ${intent?.action}", tag = TAG)
+        if (!BuildConfig.DEBUG_MODE) {
```

Comment:
I think this is redundant, since it should be handled on the Gradle level.
If you want to have a redundant check just in case, I would suggest having it in the activity's onCreate so that the whole flow is stopped as soon as possible.